### PR TITLE
[qa] Add test for max number of entries in locator

### PR DESCRIPTION
**Pull request by maflcko**
Should not be merged *before* #13907

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Thanks for @jnewbery providing early feedback.

File: test/functional/p2p_invalid_locator.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    def run_test(self):
+        node = self.nodes[0]  # convenience reference to the node
+        node.generate(1)  # Get node out of IBD
+
+        self.log.info('Test max locator size')
+        block_count = node.getblockcount()
+        for msg in [msg_getheaders(), msg_getblocks()]:
+            self.log.info('Wait for disconnect when sending {} hashes in locator'.format(MAX_LOCATOR_SZ + 1))
+            node.add_p2p_connection(P2PInterface())
```
Comment (jnewbery) [inline test/functional/p2p_invalid_locator.py]: Sorry, I was unclear in my comment previously. We still need to call `wait_for_verack()`. What I meant was that we should move that call up into `P2PInterface.peer_connect()` (a bit like the current `send_version` argument).

**Review by jnewbery — COMMENTED** [review]
One comment inline. Otherwise looks great.

**Comment by jnewbery** [comment]
Tested ACK fa310aeb0fb1840cf5197057fad6ed376e7c8ec3 (assuming #13916 goes in, otherwise I think node connection in this is racy)

**Comment by gmaxwell** [comment]
utACK

**Comment by MarcoFalke** [comment]
Force pushed new version. Only change should be 64->101

**Comment by achow101** [comment]
ACK fa85c985edbfd297aefb613445deab37c8769753

**Comment by sipa** [comment]
Concept ACK

**Comment by gmaxwell** [comment]
ACK

**Pull request merged**